src/components/App/PuzzleGenerate/index.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+import styles from './PuzzleGenerate.module.scss';
+import { Button } from '../../Global';
+import { getAllPuzzles } from '../../../services';
+import {createPuzzleInstance} from "../../../services/puzzleInstance";
+
+const PuzzleGenerate = () => {
+  const handleSubmit = event => {
+    event.preventDefault();
+    const hint = event.target[0].value;
+    const latitude = event.target[1].value;
+    const longitude = event.target[2].value;
+    const address = event.target[3].value;
+    const puzzle = event.target[4].value;
+
+    console.log(hint, latitude, longitude, address, puzzle);
+    createPuzzleInstance(puzzle, longitude, latitude, address, hint).then(r => console.log(r));
+  };
+
+  const [puzzleList, setPuzzleList] = useState([]);
+
+  useEffect(() => {
+    const fetchPuzzles = async () => {
+      setPuzzleList(await getAllPuzzles());
+    };
+    fetchPuzzles();
+  }, []);
+
+  return (
+    <>
+      <form className={styles.form} onSubmit={event => handleSubmit(event)}>
+        <h2> Make a puzzle instance </h2>
+        <input type="text" id="puzzleHint" placeholder="Hint" />
+        <div className={styles.puzzleLocation}>
+          <input type="text" id="puzzleLatitude" placeholder="Latitude" />
+          <input type="text" id="puzzleLongitude" placeholder="Longitude" />
+          <input type="text" id="puzzleAddress" placeholder="Address" />
+        </div>
+        <div>
+          {puzzleList.length > 0 && (
```

Comment:
I ran into an issue when I ran this - what if there aren't any puzzles?

An idea:
The `<select>` should show up regardless, but since there are no puzzles they won't be able to select a value, and if we make this field required then they won't be able to submit the form at all if there are no puzzles

Reply:
Also couldn't we do this in the falsey way like `puzzleList && ...`? I think I remember reading somewhere that `.length > 0` is better but I forgot why -- look into this

Reply:
how did this happened, I tried fasley way will not work since [] is not faley and I always return [] if the API did not work

Reply:
I deleted my puzzle in the database but then it disappear right away hmmmmm 🤔

Reply:
Deleted your puzzle?

Reply:
The one I created manually

Reply:
I'm not sure what you mean but the original comment is already resolved